collated/main/lincredibleJC.md

```diff
@@ -0,0 +1,772 @@
+     */
+    public static Optional<Grades> parseGrades(Optional<String> grades) throws IllegalValueException {
+        requireNonNull(grades);
+        return grades.isPresent() ? Optional.of(new Grades(grades.get())) : Optional.empty();
+    }
+```
+###### \java\seedu\address\logic\statistics\Statistics.java
+``` java
+/**
+ * Calculates statistics of the persons inside an ObservableList
+ */
+public class Statistics {
+
+    static final String NO_PERSONS_MESSAGE = "There are no persons";
+    static final String INSUFFICIENT_DATA_MESSAGE = "Insufficient Data";
+    private final int numDecimalPlace = 2;
+
+    private double[] scoreArray;
+    private int size;
+
+    public Statistics(ObservableList<ReadOnlyPerson> personList) {
+        initScore(personList);
+    }
+
+    protected Statistics(double[] scoreArray) {
+        initScore(scoreArray);
+    }
+
+    /**
+     * Takes in a PersonList and initialises the appropriate values to the Statistics instance
+     *
+     * @param personList the list of persons being taken in
+     */
+    public void initScore(ObservableList<ReadOnlyPerson> personList) {
+        int listSize = personList.size();
+        double[] listArray = new double[listSize];
+        for (int i = 0; i < listSize; i++) {
+            Person person = (Person) personList.get(i);
+            listArray[i] = Double.parseDouble((person.getGrades().value));
+        }
+        initScore(listArray);
+    }
+
+    /**
+     * Takes in an array and assigns the appropriate values to the Statistics instance
+     *
+     * @param scoreArray the array of doubles used fo calculating statistics
+     */
+    public void initScore(double[] scoreArray) {
+        Arrays.sort(scoreArray);
+        this.scoreArray = scoreArray;
+        this.size = scoreArray.length;
+    }
+
+    private double getMean() {
+        return DoubleStream.of(scoreArray).sum() / size;
+    }
+
+    public String getMeanString() {
+        if (size > 0) {
+            return getRoundedStringFromDouble(getMean(), numDecimalPlace);
+        } else {
+            return NO_PERSONS_MESSAGE;
+        }
+    }
+
+    private double getMedian() {
+        return (size % 2 == 1)
+                ? scoreArray[(size - 1) / 2]
+                : (scoreArray[(size - 1) / 2] + scoreArray[((size - 1) / 2) + 1]) / 2;
+    }
+
+    public String getMedianString() {
+        if (size > 0) {
+            return getRoundedStringFromDouble(getMedian(), numDecimalPlace);
+        } else {
+            return NO_PERSONS_MESSAGE;
+        }
+    }
+
+    private double getMedianWithIndexes(double[] arr, int startIndex, int endIndex) {
+        int currSize = endIndex - startIndex + 1;
+        return (currSize % 2 == 0)
+                ? (arr[startIndex + currSize / 2 - 1] + arr[startIndex + currSize / 2]) / 2
+                : arr[startIndex + (currSize - 1) / 2];
+    }
+
+    private double getMode() {
```

Comment:
Missing header comment. All non trivial methods should have javadoc format header comments.